addons/account/models/account_bank_statement.py

```diff
@@ -714,10 +457,111 @@ def _prepare_move_line_default_vals(self, counterpart_account_id=None):
         }
         return [liquidity_line_vals, counterpart_line_vals]
 
+    def _get_currencies(self):
+        """
+        Returns company, journal and foreign currencies of the lines
+        """
+        company_currency = self.journal_id.company_id.currency_id
+        journal_currency = self.journal_id.currency_id or company_currency
+        foreign_currency = self.foreign_currency_id or journal_currency or company_currency
+        return company_currency, foreign_currency, journal_currency
+
+    def _get_amounts(self):
+        """
+        Returns the line amount in company, journal and foreign currencies
+        """
+        company_currency, foreign_currency, journal_currency = self._get_currencies()
```

Comment:
need self.ensure_one()

Reply:
The parents (methods that are calling this),they all have ensure_one, I thought it would be useless to check it here again